pkg/server/getRecommendations.go

```diff
@@ -0,0 +1,189 @@
+	namespace := getRecommendations[0].Namespace
+	clusterName := getRecommendations[0].ClusterName
+
+	//if recommendationid provided default to use recommendation id
+	if recommendationId != "" {
+		reqParts := strings.Split(recommendationId, "_")
+		id := reqParts[3]
+
+		rec = getById(id)
+
+	}
+	//if recommendationid is missing but clustername and namespace provided:
+	if namespace != "" && clusterName != "" && recommendationId == "" {
+
+		rec = getByClusterNamespace(clusterName, namespace)
+
+		// if missing both namespace and recommendation or if missing both namespace and clustername error
+	} else if namespace == "" && recommendationId == "" || namespace == "" && clusterName == "" {
+		klog.V(5).Infof("Request missing both RecommendationId and Namespace. Need at least one to fulfill request.")
+		http.Error(w, "{\"message\":\"Both  RecommendationId and Namespace cannot be empty.\"}", http.StatusBadRequest)
+		return
+	}
+
+	client := utils.HTTPClient()
+
+	for _, request := range requestUrlList {
+
+		req, err := http.NewRequest("GET", request, nil)
+
+		if ok := helpers.ErrorHandlingRequests(w, err); !ok {
+			rec.RecommendationStatus = fmt.Sprint("Error", err)
+			return
+		}
+
+		res, err := client.Do(req)
+		if err != nil {
+			rec.RecommendationStatus = fmt.Sprint("Error", err)
+
+			klog.Errorf("Error when calling listRecommendations %v", err)
+
+		}
+
+		body, err := io.ReadAll(res.Body)
+		if err != nil {
+			rec.RecommendationStatus = fmt.Sprint("Error", err)
+
+			klog.Errorf("Error reading data from the response body %v", err)
+
+		}
+
+		if err := json.Unmarshal(body, &recommendations); err != nil {
+			rec.RecommendationStatus = fmt.Sprint("Error", err)
+
+			klog.Errorf("Cannot unmarshal response data: %v", err)
+
+		}
+		_, err = w.Write([]byte(body))
+
+		if err != nil {
+			rec.RecommendationStatus = fmt.Sprint("Error", err)
+
+			klog.Warning("Unexpected error processing the response. ", err.Error())
+			http.Error(w, "Unexpected error processing the response", http.StatusInternalServerError)
+			return
+		}
+
+		rec.RecommendationStatus = "Recieved Recommendation"
+
+		for dep, deplist := range rec.Recommendation {
+			for _, conlist := range deplist {
+				for con := range conlist {
+
+					rec = addRec(rec, dep, con, recommendations)
+
+				}
+			}
+		}
+
+		klog.Info(rec.Recommendation)
+		klog.V(4).Info("Received recommendations")
+
+	}
+}
+
+func getById(id string) *RecommendationItem {
+
+	var recitem *RecommendationItem
+	for _, rec := range Recommendationstore.data {
+		if rec.RecommendationID == id {
+			recitem = rec
+			for dep, deplist := range rec.Recommendation {
+				for _, conlist := range deplist {
+					for con := range conlist {
+
+						containerRequestUrl := fmt.Sprint(list_recommendations_url + "?" + "experiment_name=" + "ns_" + rec.Cluster + "_" + rec.Namespace + "_" + id + "-" + dep + "-" + con)
+						requestUrlList = append(requestUrlList, containerRequestUrl)
+
+					}
+				}
+			}
```

Comment:
We got the RecommendationItem to return, so it's not clear why we need this for loop after.